web/src/stream_settings_ui.ts

```diff
@@ -471,8 +493,11 @@ export function update_empty_left_panel_message(): void {
         $("#channels_overlay_container .stream-row:not(.notdisplayed)").length === 0;
     const has_search_query =
         $<HTMLInputElement>("#stream_filter input[type='text']").val()!.trim() !== "";
-    // Show "no channels match" text if all channels are hidden and there's a search query.
-    if (has_hidden_streams && has_search_query) {
+    const has_filter =
+        stream_settings_data.get_current_channel_visibility_filter() !==
+        stream_settings_data.FILTERS.ALL_CHANNELS;
+    // Show "no channels match" text if all channels are hidden and there's a search query or a filter is applied.
+    if (has_hidden_streams && (has_search_query || (has_filter && has_streams))) {
```

Comment:
We previously showed "No channels match" text even when there were no channels in the list originally and user types something in the filter input.

But now when user selects any option other than "All channels" from the dropdown, we show the "No channels match" only when there were actually some streams in that list.

I think the behavior is fine, but just wanted to make a note of it.